### PR TITLE
test (gradle-plugin/it) : Add gradle integration test for RouteEnricher (#1312)

### LINKED ISSUE #1312: Gradle-Enricher: RouteEnricher
### Description

Child of #31

RouteEnricher works with Gradle

### AC
- [x] Enricher documentation is ported to Gradle
- [x] IT `*with-route-*` are ported to Gradle
- [x] openshift.yml is generated as expected

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description
Fix #1312

+ Fix wrong property name being used in route integration tests. Looks like this property wasn't replaced during FMP to jkube migration.
+ Add gradle integration test to verify `jkube.openshift.generateRoute`
    flag works as expected.
+ Add asciidoc documentation for RouteEnricher

Signed-off-by: Rohan Kumar <rohaan@redhat.com>

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have implemented unit tests to cover my changes
 - [X] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: openshift-maven-plugin/it/src/it/simple-with-route-flag-false/invoker.properties [hunk 1]
```diff
@@ -13,5 +13,5 @@
 #
 
 invoker.goals.1=clean oc:resource
-invoker.mavenOpts=-Djkube.verbose -Djkube.offline=true -Dfabric8.openshift.generateRoute=false
+invoker.mavenOpts=-Djkube.verbose -Djkube.offline=true -Djkube.openshift.generateRoute=false
```
Comment (manusa) [inline openshift-maven-plugin/it/src/it/simple-with-route-flag-false/invoker.properties:16]: I'm really curious why this wasn't causing a test failure

**Review by manusa — APPROVED** [review]
LGTM, thx!

File: openshift-maven-plugin/it/src/it/simple-with-route-flag-false/invoker.properties (hunk 1 above)
Comment (rohanKanojia) [reply openshift-maven-plugin/it/src/it/simple-with-route-flag-false/invoker.properties:16]: Our current integration test suite doesn't seem to be verifying missing elements from expected yaml. It only verifies elements present. I had added this assertion to make sure that test fails:

https://github.com/eclipse/jkube/blob/ded7049e0ff253e04ebf1fddfc265bf25f1b636b/openshift-maven-plugin/it/src/it/simple-with-route-flag-false/verify.groovy#L23

**Pull request merged**